src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java

```diff
@@ -93,11 +93,20 @@ public Active getAuthenticated(DataFetchingEnvironment environment) throws Excep
         List<GraphRecordingDescriptor> recordings = new ArrayList<>(source.active);
 
         FilterInput filter = FilterInput.from(environment);
+        List<String> names = new ArrayList<>();
         if (filter.contains(FilterInput.Key.NAME)) {
             String recordingName = filter.get(FilterInput.Key.NAME);
+            names.add(recordingName);
+        }
+        if (filter.contains(FilterInput.Key.NAMES)) {
+            List<String> recordingNames = filter.get(FilterInput.Key.NAMES);
+            names.addAll(recordingNames);
```

Comment:
This implementation is going to behave differently from the others:

```
var names = [];
if (filters.contains(NAME_FILTER))
  names.add(filters.get(NAME_FILTER))
if (filters.containers(NAMES_FILTER))
  names.addAll(filters.get(NAMES_FILTER))
recordings = recordings.filter(r -> names.contains(r.getName())) // if r.getName() matches the NAME or any of the NAMES
```

vs

```
if (filters.contains(NAME_FILTER))
  recordings = recordings.filter(r -> r.getName().equals(filters.get(NAME_FILTER))) // if r.getName() matches the one NAME filter
if (filters.contains(NAMES_FILTER))
  recordings = recordings.filter(r -> filters.get(NAMES_FILTER)).contains(r.getName()) // if r.getName() matches any of the NAMES
```

If the client specifies both the `name: ""` and `names: []` filter inputs, we basically have that the first implementation does a set union and the second does a set intersection.

Reply:
Should we have both .matches and .
contains implemented?

Reply:
Those specific method calls are an internal detail that clients don't need to worry about. What matters is that, regardless of which type of resource they are querying for, the behaviour they get when specifying a filter with both `NAME` and `NAMES` should be consistent. It should only be done one way or the other.

Generally in cases like this, I like to follow the "principle of least surprise." A user might come across this API and not realize that there are both `NAME` and `NAMES` filters, and accidentally use one, or the other, or both. If they do that, then that behaviour should be consistent with how any other pair of filters combines.

If a user queries for active recordings and specifies a name filter as well as a label filter, what result do they get? Is it all recordings that have that name or those labels, or is it all recordings that have both that name and those labels?